File: javascript/examples/partner-api-example.js
/**
 * Example usage of the Partner API client
 *
 * Prerequisites:
 *   export ASOBA_API_KEY=your_api_key
 */
const { OnaSDK } = require('../src/index');

async function main() {
  // Initialize SDK — picks up ASOBA_API_KEY from environment
  const sdk = new OnaSDK();

  if (!sdk.config.apiKey) {
    console.error('❌ ASOBA_API_KEY environment variable not set');
    process.exit(1);
  }

  if (!sdk.partner) {
    console.error('Partner API client not initialized. Check your configuration.');
    process.exit(1);
  }

  const siteId = 'Sibaya';

  try {
    console.log(`--- Fetching KPI Rollup for ${siteId} ---`);
    const start = Date.now();
    const kpis = await sdk.partner.getKpiRollup({ site_id: siteId });
    const duration = Date.now() - start;
    console.log(`Fetch 1 took ${duration}ms`);
    console.log('Data:', JSON.stringify(kpis, null, 2));

    console.log(`\n--- Fetching KPI Rollup again (should use cache) ---`);
    const start2 = Date.now();
    const cachedKpis = await sdk.partner.getKpiRollup({ site_id: siteId });
    const duration2 = Date.now() - start2;
    console.log(`Fetch 2 took ${duration2}ms (status: ${cachedKpis ? 'OK' : 'Empty'})`);

    if (duration2 < duration) {
      console.log(
        '✅ Success: Second fetch was faster (served from cache via 304 Not Modified)'
      );
    }

    console.log(`\n--- Fetching Maintenance Signals ---`);
    const signals = await sdk.partner.getMaintenanceSignals({
      site_id: siteId,
      severity: 'high',
    });
    console.log('Signals:', JSON.stringify(signals, null, 2));

    console.log(`\n--- Fetching Forecast Snapshot ---`);
    const forecast = await sdk.partner.getForecastSnapshot({ site_id: siteId });
    console.log(
      `Horizon: ${forecast.horizon_hours}h, intervals: ${(forecast.intervals || []).length}`
    );

    console.log(`\n--- Fetching Maintenance Schedule (SEP-062) ---`);
    const schedule = await sdk.partner.getMaintenanceSchedule({ site_id: siteId });
    const summary = schedule.summary || {};
    console.log('Horizon:', JSON.stringify(schedule.horizon));
    console.log('Total tasks:', summary.total_tasks);
    console.log('By priority:', JSON.stringify(summary.by_priority));
    const tasks = schedule.tasks || [];
    if (tasks.length > 0) {
      console.log('First task:', JSON.stringify(tasks[0], null, 2));
    } else {
      console.log('No scheduled maintenance tasks yet (insufficient anomaly history).');
    }
  } catch (error) {
    console.error('Error:', error.message);
    if (error.response) {
      console.error('Response status:', error.response.status);
      console.error('Response data:', error.response.data);
    }
  }
}

main();

File: python/examples/partner_api_example.py
import json
import time

from asoba import OnaClient


def main():
    # Initialize client — picks up ASOBA_API_KEY from environment
    client = OnaClient()

    if not client.config.api_key:
        print("❌ ASOBA_API_KEY environment variable not set")
        return

    site_id = "Sibaya"

    try:
        print(f"--- Fetching KPI Rollup for {site_id} ---")
        start = time.time()
        kpis = client.partner.get_kpi_rollup(site_id=site_id)
        duration = (time.time() - start) * 1000
        print(f"Fetch 1 took {duration:.2f}ms")
        print("Data:", json.dumps(kpis, indent=2, default=str))

        print("\n--- Fetching KPI Rollup again (should use cache) ---")
        start2 = time.time()
        cached_kpis = client.partner.get_kpi_rollup(site_id=site_id)
        duration2 = (time.time() - start2) * 1000
        print(f"Fetch 2 took {duration2:.2f}ms (status: {'OK' if cached_kpis else 'Empty'})")

        if duration2 < duration:
            print("✅ Success: Second fetch was faster (served from cache via 304 Not Modified)")

        print("\n--- Fetching Maintenance Signals ---")
        signals = client.partner.get_maintenance_signals(
            site_id=site_id,
            severity="high",
        )
        print("Signals:", json.dumps(signals, indent=2, default=str))

        print("\n--- Fetching Forecast Snapshot ---")
        forecast = client.partner.get_forecast_snapshot(site_id=site_id)
        print(
            f"Horizon: {forecast.get('horizon_hours')}h, "
            f"intervals: {len(forecast.get('intervals', []))}"
        )

        print("\n--- Fetching Maintenance Schedule (SEP-062) ---")
        schedule = client.partner.get_maintenance_schedule(site_id=site_id)
        summary = schedule.get("summary", {})
        print(f"Horizon: {schedule.get('horizon')}")
        print(f"Total tasks: {summary.get('total_tasks')}")
        print(f"By priority: {summary.get('by_priority')}")
        tasks = schedule.get("tasks", [])
        if tasks:
            print(f"First task: {json.dumps(tasks[0], indent=2, default=str)}")
        else:
            print("No scheduled maintenance tasks yet (insufficient anomaly history).")

    except Exception as e:
        print(f"Error: {e}")


if __name__ == "__main__":
    main()
